src/layout/mod.rs

```diff
@@ -3745,6 +3792,60 @@ impl<W: LayoutElement> Layout<W> {
         monitor.move_workspace_up();
     }
 
+    pub fn set_workspace_name(&mut self, name: String) {
+        // also ignore the request if the name is already used by another workspace
+        if self.find_workspace_by_name(&name).is_some() {
+            return;
+        }
+
+        let Some(ws) = self.active_workspace_mut() else {
+            return;
+        };
+
+        ws.name.replace(name);
+
+        let wsid = ws.id();
+
+        // ensure last workspace is "unnamed"
```

Comment:
It's more fun than this: with `empty_workspace_above_first` option set, you need to do the same check for the *first* workspace.

Do you mind writing two unit tests which set the name of the last and the first workspace respectively? Don't need to check anything manually, just a list of operations and `check_ops()`, it should verify everything already.